ibis/backends/mysql/__init__.py

```diff
@@ -150,22 +144,20 @@ def do_connect(
             month : int32
 
         """
-        self.con = pymysql.connect(
+        self.con = MySQLdb.connect(
             user=user,
-            host=host,
+            host="127.0.0.1" if host == "localhost" else host,
```

Comment:
localhost will try to use a socket, which is different from the behavior of pymysql, so I made localhost usage backwards compatible. There's a `unix_socket` argument if anyone really needs it.